### PR TITLE
feat(contact-form): complete logic and designs

### PR DESCRIPTION
**Pull request by yasht01**
- Updated designs to the latest Figma file.
- Add `client.post()` call for sending contact form data.
- **Update tests to work with latest designs.**

### CODE REVIEW

File: lib/infrastructure/contact_form/contact_form_api.dart [hunk 1]
```diff
@@ -1,10 +1,15 @@
 import 'dart:convert';
 import 'dart:io';
 
+import 'package:collaction_app/domain/user/user.dart';
```
Comment (wizlif) [inline lib/infrastructure/contact_form/contact_form_api.dart]: Use relative imports

File: lib/infrastructure/contact_form/contact_form_api.dart [hunk 2]
```diff
@@ -13,6 +18,8 @@ import '../../domain/core/i_settings_repository.dart';
 class ContactFormApi extends IContactFormApi {
   final http.Client client;
   final ISettingsRepository settingsRepository;
+  // final _userRepository = getIt<IUserRepository>();
+  // final _authRepository = getIt<IAuthRepository>();
```
Comment (wizlif) [inline lib/infrastructure/contact_form/contact_form_api.dart]: Remove unused import

File: lib/infrastructure/contact_form/contact_form_api.dart [hunk 3]
```diff
@@ -21,19 +28,24 @@ class ContactFormApi extends IContactFormApi {
...
 
     final packageInfo = await PackageInfo.fromPlatform();
-    // ignore: prefer_interpolation_to_compose_strings
-    final bodyJson = '{"email":"${contents.email}",' +
-        // TODO add subject to ContactFormContents class
-        '"subject":"Hello CollAction!",' +
-        '"message":"${contents.message?.replaceAll('"', '\\"')}",' +
-        '"app_version":"${Platform.operatingSystem} ${packageInfo.version}+${packageInfo.buildNumber}"}';
+    // final userData = await _authRepository.getSignedInUser();
+    // final token = await userData.fold(() => null, (a) async => a.getIdToken());
```
Comment (wizlif) [inline lib/infrastructure/contact_form/contact_form_api.dart]: Remove unused imports

File: lib/presentation/contact_form/contact_form_screen.dart [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import 'package:collaction_app/presentation/shared_widgets/custom_app_bars/custom_appbar.dart';
+import 'package:collaction_app/presentation/shared_widgets/pill_button.dart';
```
Comment (wizlif) [inline lib/presentation/contact_form/contact_form_screen.dart:2]: Use relative imports

**Review by wizlif — CHANGES_REQUESTED** [review]

File: lib/infrastructure/contact_form/contact_form_api.dart [hunk 5]
```diff
@@ -21,18 +21,20 @@ class ContactFormApi extends IContactFormApi {
     assert(contents.message != null);
 
     final packageInfo = await PackageInfo.fromPlatform();
-    // ignore: prefer_interpolation_to_compose_strings
-    final bodyJson = '{"email":"${contents.email}",' +
-        // TODO add subject to ContactFormContents class
-        '"subject":"Hello CollAction!",' +
-        '"message":"${contents.message?.replaceAll('"', '\\"')}",' +
-        '"app_version":"${Platform.operatingSystem} ${packageInfo.version}+${packageInfo.buildNumber}"}';
+    final bodyJson =
```
Comment (Xazin) [inline lib/infrastructure/contact_form/contact_form_api.dart:24]: Let's make a DTO model for this using Freezed.

File: lib/infrastructure/contact_form/contact_form_api.dart [hunk 6]
```diff
@@ -21,18 +21,20 @@ class ContactFormApi extends IContactFormApi {
...
+    };
 
     return client
         .post(
-            Uri.parse('${await settingsRepository.baseApiEndpointUrl}/contact'),
-            encoding: Encoding.getByName('application/json'),
-            body: bodyJson)
+          Uri.parse('${await settingsRepository.baseApiEndpointUrl}/contact'),
+          encoding: Encoding.getByName('application/json'),
+          body: bodyJson,
```
Comment (Xazin) [inline lib/infrastructure/contact_form/contact_form_api.dart:35]: Use the `toJson` from the ContactFormDto class once you've created it.

File: lib/presentation/contact_form/contact_form_screen.dart [hunk 7]
```diff
@@ -86,37 +83,69 @@ class ContactFormPageState extends State<ContactFormPage> {
...
+                                borderRadius: BorderRadius.circular(20.0),
+                                borderSide: const BorderSide(
+                                    width: 0, color: Colors.transparent),
+                              ),
+                            ),
+                          ),
+                          Padding(
+                            padding: const EdgeInsets.fromLTRB(16, 4, 0, 0),
+                            child: Text(
+                              "We will send out response to your email address",
```
Comment (Xazin) [inline lib/presentation/contact_form/contact_form_screen.dart:108]: `out` => `our`

**Comment by Xazin** [comment]
@wizlif @yasht01 
Please review again, check the commit I pushed on top.

**Comment by Xazin** [comment]
Closed due to #78

**Pull request closed without merging**